stage0/gc.go

```diff
@@ -65,49 +67,111 @@ type mount struct {
 	id         int
 	parentId   int
 	mountPoint string
+	mode       string
 }
 
 type mounts []*mount
 
-// Less ensures that mounts are sorted in an order we can unmount; child before
-// parent. This works by pushing all child mounts up to the front. Parent
-// mounts are halted once they encounter a child. The actual sequential order
-// of the mounts doesn't matter; only that children are in front of parents.
-func (m mounts) Less(i, j int) bool { return m[i].id != m[j].parentId }
-func (m mounts) Len() int           { return len(m) }
-func (m mounts) Swap(i, j int)      { m[i], m[j] = m[j], m[i] }
+// getAncestors searches the list for all ancestors of the mount at index i
+func (m mounts) getAncestors(i int) map[int]struct{} {
+	ancestorMap := make(map[int]struct{}, 0)
+	current := m[i]
+	for found := true; found; {
+		found = false
+		for _, mnt := range m {
+			if mnt.id == current.parentId {
+				ancestorMap[mnt.id] = struct{}{}
+				current = mnt
+				found = true
+				break
+			}
+		}
+	}
+	return ancestorMap
```

Comment:
Since `getAncestors` is only used to get the number of parents, you could return an `int` instead of a `set`.